R/tar-terra-rasts.R

```diff
@@ -0,0 +1,160 @@
+                           resources = targets::tar_option_get("resources"),
+                           storage = targets::tar_option_get("storage"),
+                           retrieval = targets::tar_option_get("retrieval"),
+                           cue = targets::tar_option_get("cue")) {
+  check_pkg_installed("terra")
+
+  name <- targets::tar_deparse_language(substitute(name))
+
+  envir <- targets::tar_option_get("envir")
+
+  command <- targets::tar_tidy_eval(
+    expr = as.expression(substitute(command)),
+    envir = envir,
+    tidy_eval = tidy_eval
+  )
+
+  pattern <- targets::tar_tidy_eval(
+    expr = as.expression(substitute(pattern)),
+    envir = envir,
+    tidy_eval = tidy_eval
+  )
+
+  drv <- get_gdal_available_driver_list("raster")
+
+  # if not specified by user, pull the corresponding geotargets option
+  filetype <- filetype %||% geotargets_option_get("gdal.raster.driver")
+  filetype <- rlang::arg_match0(filetype, drv$name)
+
+  gdal <- gdal %||% geotargets_option_get("gdal.raster.creation_options")
+
+  targets::tar_target_raw(
+    name = name,
+    command = command,
+    pattern = pattern,
+    packages = packages,
+    library = library,
+    format = create_format_terra_rasters_sprc(filetype = filetype,
+                                              gdal = gdal,
+                                              ...),
+    repository = repository,
+    iteration = iteration,
+    error = error,
+    memory = memory,
+    garbage_collection = garbage_collection,
+    deployment = deployment,
+    priority = priority,
+    resources = resources,
+    storage = storage,
+    retrieval = retrieval,
+    cue = cue
+  )
+}
+
+#' @param filetype File format expressed as GDAL driver names passed to `terra::writeRaster()`
+#' @param gdal GDAL driver specific datasource creation options passed to `terra::writeRaster()`
+#' @param ... Additional arguments not yet used
+#' @noRd
+create_format_terra_rasters_sprc <- function(filetype, gdal, ...) {
+  check_pkg_installed("terra")
+
+  drv <- get_gdal_available_driver_list("raster")
+
+  filetype <- filetype %||% geotargets_option_get("gdal.raster.driver")
+  filetype <- rlang::arg_match0(filetype, drv$name)
+
+  gdal <- gdal %||% geotargets_option_get("gdal.raster.creation_options")
+  ## TODO
+  ## Need to append the "opt" argument for GDAL options that is currently
+  ## controlled with the if(i > 1) part.
+
+  .write_terra_rasters_sprc <- eval(
+    substitute(
+      function(object, path) {
+        for (i in seq(object)) {
+          if (i > 1) {
+            opt <- "APPEND_SUBDATASET=YES"
+          } else {
+            opt <- ""
+          }
+          terra::writeRaster(
+            x = object[i],
+            filename = path,
+            filetype = filetype,
+            overwrite = (i == 1),
+            gdal = opt
+          )
+        }
+      },
+      list(filetype = filetype, gdal = gdal)
+    )
+  )
```

Comment:
I wonder if it might be more useful to pull this out and create a `write_sprc()` function?  Would it work on any SpatRasterCollection outside of a targets pipeline?

Reply:
I'm not sure if this is something we should add, in terms of reading/writing function outside of geotargets, we should let the targets manage that work.